llama-rs/src/lib.rs

```diff
@@ -147,6 +147,16 @@ type Token = String;
 pub struct Vocabulary {
     /// Maps every integer (index) token id to its corresponding string
     mapping: Vec<Token>,
+    scores: Vec<f32>,
+}
+
+impl Vocabulary {
+    fn with_capacity(n_vocab: usize) -> Self {
+        Self {
+            mapping: Vec::<Token>::with_capacity(n_vocab),
+            scores: Vec::<f32>::with_capacity(n_vocab),
+        }
+    }
```

Comment:
Unfortunately there's now a conflict between this and the recently merged #39. Shouldn't be hard to fix, but let me know if you have questions!